test/unit-test/test_lib/test_analysis/test_user.py

```diff
@@ -1,34 +1,135 @@
+        expected_keywords_filtered = util.load_from_disk(current_directory + "/data/user_test/keywords/keywords_filtered")
+        expected_user_keyword_freq_dict = util.load_from_disk(current_directory + "/data/user_test/user_keyword_freq_dict")
+        expected_user_words_dict = util.load_from_disk(current_directory + "/data/user_test/keywords/user_words_dict") 
+        expected_nicks_for_stop_words = util.load_from_disk(current_directory + "/data/user_test/keywords/nicks_for_stop_words")
+        expected_sorted_keywords_for_channels = util.load_from_disk(current_directory +"/data/user_test/keywords/sorted_keywords_for_channels") 
+
+        assert expected_keywords_filtered == keywords_filtered
+        assert expected_user_keyword_freq_dict == user_keyword_freq_dict
+        assert expected_user_words_dict == user_words_dict
+        assert expected_nicks_for_stop_words == nicks_for_stop_words
+        assert expected_sorted_keywords_for_channels == sorted_keywords_for_channels
+
+
+    @patch("lib.analysis.user.keywords", autospec = True)
+    @patch("lib.analysis.user.extended_stop_words", autospec = True)
+    @patch("lib.analysis.user.time", autospec = True)
+    def test_keywords_clusters(self,mock_time, mock_extended_stop_words, mock_keywords):
+        keywords_filtered = util.load_from_disk(current_directory + "/data/user_test/keywords/keywords_filtered")
+        user_keyword_freq_dict = util.load_from_disk(current_directory + "/data/user_test/user_keyword_freq_dict")
+        user_words_dict = util.load_from_disk(current_directory + "/data/user_test/keywords/user_words_dict") 
+        nicks_for_stop_words = util.load_from_disk(current_directory + "/data/user_test/keywords/nicks_for_stop_words")
+        sorted_keywords_for_channels = util.load_from_disk(current_directory +"/data/user_test/keywords/sorted_keywords_for_channels") 
+
+
+        mock_keywords.return_value = keywords_filtered, user_keyword_freq_dict, user_words_dict, nicks_for_stop_words, sorted_keywords_for_channels
+        mock_extended_stop_words.side_effect = util.load_from_disk(current_directory + "/data/user_test/extended_stop_words_list")
+        mock_time.return_value = 0 
+        captured_output = StringIO.StringIO()
+        sys.stdout = captured_output
+        user.keywords_clusters(self.log_data, self.nicks_hash, self.nick_same_list, current_directory + "/data/user_test/","temp_output_keywords_clusters")
+        sys.stdout = sys.__stdin__
+        f = open(current_directory + "/data/user_test/stdout_captured_output_keywords_clusters.txt", "r")    
+        expected_captured_output = f.read()
+        f.close()
+        output = captured_output.getvalue() 
+        assert expected_captured_output == output 
+        assert filecmp.cmp(current_directory + "/data/user_test/output_keywords_clusters.txt", current_directory + "/data/user_test/temp_output_keywords_clusters.txt")
```

Comment:
@kaushik-rohit and @bhuvangupta008 This pattern of matching seems fairly readable. Instead of reading from file into an object and comparing with the object generated, we can also put the object value into a file using util.save_to_disk() function and then on both the files we can apply filecmp.cmp() function.

Obviously, when we are performing file comparison this method makes more sense. When we are doing object comparison, doing object equality makes more sense. We can acknowledge the use of this file comparison pattern and put it in coding style document.

@bhuvangupta008 you can update the coding style document.